### PR TITLE
refactor: use stubs for gdk-pixbuf dependency

### LINKED ISSUE #34060: [Bug]: libgdk-pixbuf library is not dynamically linked
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

17.x.y

### What operating system are you using?

Ubuntu

### Operating System Version

Xubuntu 20.04, Ubuntu 16.04

### What arch are you using?

x64

### Last Known Working Electron version

_No response_

### Expected Behavior

While porting over Chromium's script to generate dependencies for Debian packages in VS Code (https://github.com/microsoft/vscode/issues/13089), I noticed that libgdk-pixbuf is statically linked rather than dynamically linked. An issue is that the exact name of the library changed over time, so sometimes it is libgdk-pixbuf-2.0-0, and other times it is libgdk-pixbuf2.0-0. Therefore, Electron should look up the library at runtime.

### Actual Behavior

Using the Electron sysroot when searching for Debian dependencies, it seems libgdk-pixbuf-2.0-0 is statically linked: https://github.com/microsoft/vscode/pull/147335/files#diff-b84879f8920b8ee14b70af14c40c90a255472a38795104384f64db32f1c7505dR56

I verified in a separate CI run that the dependency is indeed contributed by Electron, and not by other shared libraries that VS Code uses.

### Testcase Gist URL

_No response_

### Additional Information

Since libgdk is responsible for rendering menus and file dialogs, I think we should log a message and crash the program if the dependency cannot be found during runtime.

CC @deepak1556

### PR DESCRIPTION
**Pull request by rzhao271**
#### Description of Change

Fixes #34060

While adding automatic dependency generation for Debian packages in VS Code (https://github.com/microsoft/vscode/pull/147335), I learned that Electron statically links to libgdk-pixbuf. But, on some versions of Ubuntu, the package is called libgdk-pixbuf-2.0-0, and on other versions, it's called libgdk-pixbuf2.0-0. Therefore, Electron should dynamically look for libgdk-pixbuf rather than statically link to the package, since its name varies depending on the Linux distro version.

This PR is based on #33650, which does something similar for libgtk in Electron.
Also, Chromium has already done something similar for their dependency on libgdk_pixbuf: https://source.chromium.org/chromium/chromium/src/+/3a5f2968e096ebe1faeb156510193e87f07980c9

CC @deepak1556 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: On Linux, changed the libgdk-pixbuf dependency to be dynamically linked rather than statically linked.

### CODE REVIEW

File: patches/chromium/.patches [hunk 1]
```diff
@@ -111,3 +111,4 @@ make_gtk_getlibgtk_public.patch
 build_disable_print_content_analysis.patch
 custom_protocols_plzserviceworker.patch
 feat_filter_out_non-shareable_windows_in_the_current_application_in.patch
+make_gtk_getlibgdkpixbuf_public.patch
```
Comment (deepak1556) [inline patches/chromium/.patches:114]: Can you merge this to the `make_gtk_getlibgtk_public.patch`, the changes are not unrelated.

File: shell/browser/electron_browser_main_parts.cc [hunk 2]
```diff
@@ -371,6 +371,11 @@ void ElectronBrowserMainParts::ToolkitInitialized() {
     electron::UninitializeElectron_gtk();
   }
 
+  electron::InitializeElectron_gdk_pixbuf(gtk::GetLibGdkPixbuf());
+  if (!electron::IsElectron_gdk_pixbufInitialized()) {
+    electron::UninitializeElectron_gdk_pixbuf();
+  }
```
Comment (deepak1556) [inline shell/browser/electron_browser_main_parts.cc:377]: This should be a hard check unlike the previous gtk symbols loaded there is no fallback for the gdk_pixbuf calls. Something like the following,

```suggestion
  CHECK(electron::IsElectron_gdk_pixbufInitialized()) << "Fail to load libgdk_pixbuf-2.0.so.0";
```

**Review by nornagon — APPROVED** [review]
seems legit

**Review by deepak1556 — APPROVED** [review]

**Comment by deepak1556** [comment]
Failing tests are unrelated https://github.com/electron/electron/issues/34575, merging.

**Pull request merged**